src/SimulationRunner.cc

```diff
@@ -536,6 +536,11 @@ void SimulationRunner::UpdateSystems()
     IGN_PROFILE("Update");
     for (auto& system : this->systemsUpdate)
       system->Update(this->currentInfo, this->entityCompMgr);
+
+    if (serverConfig.SameProcessAsGUI())
+    {
+      this->eventMgr.Emit<events::UpdateSystems>();
```

Comment:
I think this is not that easy because of the ECM.  we can't lose any step in the loop that include any change in the ECM, otherwise gui plugins are not aware of these changes. For these reason I emitted the event UpdateSystem, which internally will be only executed if there is any change in the ECM

https://github.com/ignitionrobotics/ign-gazebo/blob/47a87eb14d204a7590a218497aac1d88dab594d6/src/gui/GuiRunner.cc#L253-L261


I have added a pool of workers to update the GUI https://github.com/ignitionrobotics/ign-gazebo/pull/793/commits/dc02d76d13359ad5097d74c0da0d2017e3fb6dbf . What do you think about this @chapulina @iche033  ?

Reply:
I think I understand why it's best to trigger the GUI updates before `PostUpdate`. I documented it on #975